test/e2e/test/elasticsearch/checks_k8s.go

```diff
@@ -118,6 +120,30 @@ func CheckESVersion(b Builder, k *test.K8sClient) test.Step {
 	}
 }
 
+// CheckESReady retrieves ES phase for given ES,
+// and checks if it's in Ready phase twice with 5 seconds between the checks, until success
+func CheckESReady(b Builder, k *test.K8sClient) test.Step {
+	return test.Step{
+		Name: "ES should eventually be ready",
+		Test: test.Eventually(func() error {
+			// loop helps win the race condition between this check and operator changing the phase
+			for i := 0; i < 2; i++ {
+				var es estype.Elasticsearch
+				if err := k.Client.Get(k8s.ExtractNamespacedName(&b.Elasticsearch), &es); err != nil {
+					return err
+				}
+
+				if es.Status.Phase != estype.ElasticsearchReadyPhase {
+					return fmt.Errorf("ES is not in Ready phase yet")
+				}
+
+				time.Sleep(5 * time.Second)
```

Comment:
why waiting 5 seconds and not relying on `test.Eventually ` default retry delay? Also not sure why `2`.

Reply:
There is a race condition between test code checking this condition and operator actually changing the phase to `ApplyingChanges`. If operator wins, it's all good and we keep on checking the phase until it's `Ready`. If test code (this method) wins, it's still `Ready` (didn't move to `ApplyingChanges` yet), so we just go with next checks. Sleep gives the operator 5 seconds to update the state. Two loops are there to avoid sleeping in each loop. We might already have a primitive in test to achieve the same thing, but I wasn't able to find it.

The comment above the loop was supposed to convey that, any suggestions how to word it better?

Also, both this and the change in topology check will allow for the test to pass on their own, so we might decide we just pick one of those two changes. But it seems to me that this is also something that we want to check before going forward.

Reply:
We chatted offline and the conclusion was to rely on topology check for making sure changes are applied as expected. ES ready check can be added after that (although in a separate PR). Also, it was noted that many other checks hit the same race condition issue as this does. It will be tracked under #1825.